privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -344,7 +346,12 @@ def checkPass(self, uid, password):
                 return False
             # we need to check credentials with kerberos differently since we
             # can not use bind for every user
-            name = gssapi.Name(self.getUserInfo(uid).get('username'))
+            if self.getUserInfo(uid).get('upn') is not None \
```

Comment:
please assign `self.getUserInfo(uid).get('upn')` to a variable, so it is only done once and improves readability